packages/nimble-components/src/table/tests/table.stories.ts

```diff
@@ -23,10 +23,16 @@ const simpleData = [
     }
 ] as const;
 
+const simpleDataIdFieldName = 'myStr';
+
 const dataSets = {
     [ExampleDataType.simpleData]: simpleData
 } as const;
 
+const dataSetIdFieldNames = {
+    [ExampleDataType.simpleData]: simpleDataIdFieldName
+} as const;
+
 const overviewText = 'The `nimble-table` is a component that offers a way to render tabular data in a variety of ways in each column.';
 
 const dataDescription = `\`data\` is a property that is an array of records. A record provides the data that backs a single row in the table.
```

Comment:
Thoughts on documenting `id-field-name` and the new validity APIs in storybook? The client can't set all of them so they might not make sense as controls, but I can't think of a better place to make clients aware of them.

In a perfect world our docs would provide both normal API info (data types, purpose of the API) and also info about why someone might choose to set a custom ID field and when we'd recommend they perform validity checks. If we don't know the answers to all of those yet we can start with what we do know.

Reply:
I've documented the behavior `id-field-name`, `validity`, and `checkValidity()`, but I didn't add any usage recommendations. I think we'll need to work on defining what our recommendations are.